packages/beacon-node/test/unit/execution/engine/utils.test.ts

```diff
@@ -0,0 +1,191 @@
+import {expect} from "chai";
+import {ErrorAborted} from "@lodestar/utils";
+import {ExecutePayloadStatus, ExecutionEngineState} from "../../../../src/execution/index.js";
+import {
+  HTTP_CONNECTION_ERROR_CODES,
+  HTTP_FATAL_ERROR_CODES,
+  getExecutionEngineState,
+} from "../../../../src/execution/engine/utils.js";
+import {QueueError, QueueErrorCode} from "../../../../src/util/queue/errors.js";
+import {ErrorJsonRpcResponse, HttpRpcError} from "../../../../src/eth1/provider/jsonRpcHttpClient.js";
+
+describe("execution/engine/utils", () => {
+  describe("getExecutionEngineState", () => {
```

Comment:
These are very standard error codes must comply with any `fetch` implementation. Testing regressively native implementations is not suggested. But for node/browser compatibility of fetch we should have some basic tests.

Reply:
Yes, I doubt the error codes will ever change, I just tested different cases in regard to custom `FetchError` added in https://github.com/ChainSafe/lodestar/pull/5811 but these are rather meant to test if `FetchError` works as expected.

Since we use native fetch now it is even more unlikely we get unexpected errors. As per my understanding the codes you are checking right now for EL offline/auth failed should be alright and it is anyhow hard produce specific error codes (https://github.com/ChainSafe/lodestar/pull/5811#issuecomment-1658556984).